zipline/finance/performance/period.py

```diff
@@ -143,14 +143,18 @@ def __init__(
         self.asset_finder = asset_finder
         self.data_frequency = data_frequency
 
+        # Start and end of the entire period
         self.period_open = period_open
         self.period_close = period_close
 
+        self.initialize(starting_cash=starting_cash,
```

Comment:
Mainly code duplication. We have to reset the values of some attributes when we roll over a period. It was previous done this way too, but it was `rollover` that initialized some values instead